.github/workflows/release.yml

```diff
@@ -0,0 +1,163 @@
+name: Release
+
+on:
+  pull_request:
+    branches:
+      - '*'  # must quote since "*" is a YAML reserved character; we want a string
+    paths-ignore:
+      - '.github/workflows/quarto-render.yml'
+      - '_quarto.yml'
+      - 'quarto-materials/*'
+      - '**/.md'
+      - 'tiledb/doxygen/source/*'
+      - 'tiledb/sm/c_api/tiledb_version.h'
+  push:
+    branches:
+      - dev
+      - 'release-*'
+    tags:
+      - '*'
+
+jobs:
+  Build-Release:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [macos-arm64, macos-x86_64, linux-x86_64, linux-x86_64-noavx2, windows-x86_64]
+        include:
+          - platform: windows-x86_64
+            os: windows-2019
+            triplet: x64-windows-release
+          - platform: linux-x86_64
+            os: ubuntu-20.04
+            manylinux: true
+            triplet: x64-linux-release
+          - platform: linux-x86_64-noavx2
+            os: ubuntu-20.04
+            cmake_args: -DCOMPILER_SUPPORTS_AVX2=OFF
+            triplet: x64-linux-release
+            manylinux: true
+          - platform: macos-x86_64
+            os: macos-12
+            # By default we have to target macOS 10.14 at minimum according to CRAN policy.
+            # See https://stat.ethz.ch/pipermail/r-package-devel/2023q4/010078.html
+            MACOSX_DEPLOYMENT_TARGET: 10.14
+            triplet: x64-osx-release
+          - platform: macos-arm64
+            os: macos-12
+            cmake_args: -DCMAKE_OSX_ARCHITECTURES=arm64
+            MACOSX_DEPLOYMENT_TARGET: 11
+            triplet: arm64-osx-release
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.manylinux && 'quay.io/pypa/manylinux2014_x86_64:2023-11-13-f6b0c51' || '' }}
+    env:
+      MACOSX_DEPLOYMENT_TARGET: ${{ matrix.MACOSX_DEPLOYMENT_TARGET }}
+      VCPKG_BINARY_SOURCES: 'clear;x-gha,readwrite'
+
+    steps:
+      - name: Checkout TileDB
+        # v4 uses node 20 which is incompatible with the libc version of the manylinux image
+        uses: actions/checkout@v3
+      - name: Export GitHub Actions cache variables
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
+            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '');
+      - name: Set variables
+        id: get-values
+        run: |
+          release_hash=$( echo ${{ github.sha }} | cut -c-8 - )
```

Comment:
Breadcrumb for later: this switched from 7-nybble hashes in filenames to 8-nybble hashes which broke some dependent packages. See #4599 for a fix.

Reply:
https://app.shortcut.com/tiledb-inc/story/38683